### ISSUE TITLE
Further problems with authentication

### ISSUE THREAD

**Issue by csillag** [hypothesis/h#1210]
I have been testing the test extension. I see the following problems:
- [ ] When I give an invalid uname / password combo, I see two identical "Invalid username or password" messages displayed on the top. I remember adding workarounds for this; looks like latest changes broke this again.
- [ ] When I log in after a failed attempt, I get the following error message displayed in red on the top: "Your submission is invalid". But I am logged in anyway.
- [ ] When I log out after all this, I see this displayed, in green: "You are now logged in."
- [ ] I also remember the password recovery has problems, too.

In short, this is not ready for 0.1.

I am making this a blocker.

**Update:** See the updated bug list below.
